src/output-json-smtp.c

```diff
@@ -85,9 +85,8 @@ static int JsonSmtpLogger(ThreadVars *tv, void *thread_data, const Packet *p, Fl
     EveSmtpDataLogger(f, state, tx, tx_id, jb);
     jb_close(jb);
 
-    if (EveEmailLogJson(jhl, jb, p, f, state, tx, tx_id) == TM_ECODE_OK) {
-        OutputJsonBuilderBuffer(jb, jhl->ctx);
-    }
+    EveEmailLogJson(jhl, jb, p, f, state, tx, tx_id);
```

Comment:
Do we get an empty `email: {}` in the eve in such a case now?

Reply:
No. `EveEmailLogJson` reverts the opening of that object on failure. Its like it was designed to be done this way ;)